File: Hashmap/Word Pattern/WordPattern.java
import java.util.HashMap;

class Solution {
    public boolean wordPattern(String pattern, String s) {
        String[] array = s.split(" ");

        if (pattern.length() != array.length)
            return false;

        HashMap<Character, String> map = new HashMap<>();

        for (int i = 0; i < pattern.length(); i++) {
            char currentChar = pattern.charAt(i);
            if (map.containsKey(currentChar)) {
                if (!map.get(currentChar).equals(array[i])) {
                    return false;
                }
            } else {
                if (map.containsValue(array[i])) {
                    return false;
                }
            }
            map.put(currentChar, array[i]);
        }
        return true;
    }
}

File: Hashmap/Word Pattern/WordPattern.py
class Solution:
    def wordPattern(self, pattern: str, s: str) -> bool:
        if not str or not pattern:
            return False
        strings = s.split()
        if len(strings) != len(pattern):
            return False
        str2pat = {}
        pat2str = {}
        for i in range(len(pattern)):
            if strings[i] in str2pat:
                if str2pat[strings[i]] != pattern[i]:
                    return False
            else:
                str2pat[strings[i]] = pattern[i]
            if pattern[i] in pat2str:
                if pat2str[pattern[i]] != strings[i]:
                    return False
            else:
                pat2str[pattern[i]] = strings[i]
        return True
